client/src/components/QualityManagement/Parts.jsx

```diff
@@ -0,0 +1,328 @@
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+      setSelected([]);
+      refetch();
+    } catch (err) {
+      toast({
+        title: 'Error',
+        description: err.message,
+        status: 'error',
+        duration: 9000,
+        isClosable: true,
+      });
+    }
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setIsLoadingButton(true);
+    try {
+      await createDefect({
+        partName: partName,
+        type: type.current.value,
+        status: status.current.value,
+        description: description.current.value,
+        onCloseComplete: onClose,
+      });
+      toast({
+        title: 'Request Sent',
+        description: 'Defect has been created successfully',
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+      refetch();
+    } catch (err) {
+      toast({
+        title: 'Error',
+        description: err.message,
+        status: 'error',
+        duration: 9000,
+        isClosable: true,
+      });
+    }
+    setIsLoadingButton(false);
+  };
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
+  const addDefectForm = () => (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Add a new Defect</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody pb={6}>
+          <FormControl isRequired>
+            <FormLabel>Part Name</FormLabel>
+            <Input onChange={handlePartNameInput} value={partName} placeholder="Part name" />
+          </FormControl>
+
+          <FormControl mt={4}>
+            <FormLabel>Defect Type</FormLabel>
+            <Select ref={type}>
+              <option>Broken</option>
+              <option>Reparable</option>
+            </Select>
+          </FormControl>
+          <FormControl mt={4}>
+            <FormLabel>Defect Status</FormLabel>
+            <Select ref={status}>
+              <option>Solved</option>
+              <option>Pending</option>
+              <option>Ongoing</option>
+            </Select>
+          </FormControl>
+          <FormControl mt={4}>
+            <FormLabel>Description</FormLabel>
+            <Textarea ref={description} placeholder="Description" />
+          </FormControl>
+        </ModalBody>
+
+        <ModalFooter>
+          <Button
+            isLoading={isLoadingButton}
+            disabled={partName ? false : true}
+            onClick={handleSubmit}
+            colorScheme="blue"
+            mr={3}
+          >
+            Add
+          </Button>
+          <Button onClick={onClose}>Cancel</Button>
+        </ModalFooter>
+      </ModalContent>
+    </Modal>
+  );
+
+  if (isSuccess && data.data.length === 0) {
+    return (
+      <>
+        <Heading size="xl" textAlign="center" mt={5}>
+          No Defects
+        </Heading>
+        {addDefectForm()}
+        <Link onClick={onOpen}>
```

Comment:
When there are no defects, should we have an explicit button to add a defect rather than clicking on the image?

Reply:
I tried to do that but couldn't find a way to make it look nice. If you have an example, send it to me and I'll change it cuz I don't really like clicking on the image either.

Reply:
I feel that the add defect form might be hard to find for users since I didn't found the form initially and I have to look at the code to find it

Reply:
I think it's better now, I added a button